src/atomWithHash.ts

```diff
@@ -41,19 +45,22 @@ export function atomWithHash<Value>(
         window.removeEventListener('hashchange', callback);
       };
     });
-  const setHash = (searchParams: URLSearchParams) => {
-    if (options?.replaceState) {
+  const setHashOption = options?.setHash;
```

Comment:
I would also add a warning for JS users. What do you think?
```suggestion
  if (options?.replaceState) {
    console.warn('[DEPRECATED] Use setHash=replaceState instead');
  }
  const setHashOption = options?.setHash;
```

Reply:
Good point @dai-shi. I committed your suggestion.